enhancements/olm-admission-webhooks.md

```diff
@@ -0,0 +1,135 @@
+  - Disabling OLM Management of cert creation and rotation.
+  - Specifing required annotations that must be added to APIServices.
+
+### Non-Goals
+
+- OLM will not provide support for conversion webhooks at this time.
+
+## Proposal
+
+This propsal outlines a plan that enables OLM to support operators that include validating or mutating admission webhooks as a part of the operator bundle. When this plan is implemented, operator authors will be able to:
+- Ship their operator with a validating or mutating webhook without implementing logic to create or manage cert information as OLM will take care of that requirement instead.
+
+
+Likewise, Cluster Admins will be able to:
+- Disable OLM management of certificates and outsource this to a tool such as the service-ca controller or cert-manager.
+
+Before we discuss cert generation or rotation, let's do a quick review of resources created as a part of a Webhook:
+- Webhook
+- Service
+- Deployment
+- RBAC
+- Certs (Present in the Webhook and Deployment)
+
+Currently, operator authors can already define the Deployment and RBAC resources in the CSV. Furthermore, OLM could be updated to generate the service required to forward requests intercepted by the webhook to the appropriate deployment and could manage on-cluster cert rotation. This leaves the operator author with responsibility to define the webhook itself within the CSV.
+
+In order to support both validating and mutating webhooks OLM will need to provide operator authors with the ability to:
+- Specify all fields in the [ValidatingWebhook struct](https://github.com/kubernetes/api/blob/b3bd583303d6e0723bbd276c4b00d1b65c1ff8db/admissionregistration/v1/types.go#L173-L300)
+- Specify all fields in the [MutatingWebhook struct](https://github.com/kubernetes/api/blob/b3bd583303d6e0723bbd276c4b00d1b65c1ff8db/admissionregistration/v1/types.go#L302-L447)
+
+Close examination of the `ValidatingWebhook` and `MutatingWebhook` structs reveals that the `MutatingWebhook` struct contains all fields featured in the `ValidatingWebhook` struct and an additional `ReinvocationPolicy` field. As such, the `CSV` struct should be updated to provide users with the ability to define all fields in a `MutatingWebhook` struct.
+
+### CSV support for Validating and Mutating Webhook
+
+CSV support for webhooks will begin with the introduction of the `OLMWebhook` and `WebhookDescription` structs. These structures provides OLM with sufficient information to create either mutating or validating admission webhooks. These structs should look similar to the following examples:
+
+```go
+// OLMWebhook is an exact replica of the MutatingWebhook struct with the excpetion that the WebhookClientConfig is missing, as OLM will generate this resource itself.
```

Comment:
I don't think we need to prefix with `OLM` - the package names will indicate that it's a part of OLM's types

Reply:
Good point - addressed in the latest version of the PR.